src/cpp/include/UTIL/CheckCollections.h

```diff
@@ -0,0 +1,79 @@
+#ifndef CheckCollections_h
+#define CheckCollections_h 1
+
+#include "lcio.h"
+
+#include <string>
+#include <unordered_map>
+#include <set>
+
+namespace UTIL {
+
+  
+/** Utility class for checking and patching events with respect to collections that are not present 
+ *  in every event of a set of files. 
+ *  
+ *  \author F.Gaede, DESY
+ *  \date Dec 2022
+ */
+  class CheckCollections {
+    
+    typedef std::vector<std::pair<std::string,std::string>> Vector ;
+    
+  public: 
+    
+    /** Convenient c'tor.
+     */
+    CheckCollections() = default ;
+
+    /** d'tor */
+    virtual ~CheckCollections() = default ;
+
+    /** Checks the file for missing collections - can be called repeadedly on different files.
+     */
+    void checkFile( const std::string& fileName ) ;
+
+    /** Checks all files for missing collections.
+     */
+    void checkFiles( const std::vector<std::string>& fileNames ) ;
+
+    
+    /** dump result of check to stream */
+    void print(  std::ostream& os ) const ;
+    
+
+    /** Returns the collections that are not present in all events checked with checkFiles() with their names and types.
+     */
+    Vector getMissingCollections() const ;
+
+    /** Returns the collections that are present in all events checked with checkFiles() with their names and types.
+     */
+    Vector getConsistentCollections() const ;
+
+    /** Add a collection with (name,type) that should be added to events in patchEvent().
+     */
+    void addPatchCollection(const std::string name, std::string type){
+      _patchCols.push_back( {name, type} ) ;
```

Comment:
```suggestion
      addPatchCollections({{name, type}});
```
Not sure if this initializer list works, but to avoid having to duplicate the deduplication logic, I would just forward this to the overload taking the vector.

Reply:
If you really would want to put such a check in, I'd do it the other way round, i.e. check in the individual `addPatchCollection` for duplicates and use this in the other function. But I think this is really overdoing it, as duplicates would not cause any harm (except tiny extra runtime). Much more critical would be to check for viable LCIO type names as that would cause crashes. Not sure we need this for this 'expert' tool, though...

Reply:
The main reason I pointed this out is the changed behavior wrt the previous version that had a `set`. In any case, this will most likely only be called once in any case, and with an empty `_patchCols` as well, so I agree, deduplicating here is probably not necessary.

The LCIO type check is something we could/should actually do (also to catch typos in steering files), I think. It depends a bit on whether the crash is an exception or a segmentation fault in the collection creation factory. The latter might be hard to diagnose if there is no backtrace. The former, I would consider "good enough" for this, as we will have something printed to the screen with enough information, I think.

Reply:
I guess any container will do here, as all we need here is to get all elements. I had put in the vector as you suggested for further downstream use in the converter. Thinking about the type checking, again: this also is not needed or even useful to have here. We should get neither crash or exception, as the type string is really just used as 'information' and no code should depend on it. For empty collection no one will (or should) access this ever.  Collections with incorrect type might get lost in routines that list/dump collections as `LCTOOLS::dumpEventDetailed` - but no crashes.

Reply:
Removed the 'WIP', so from my side it would be good to go.